Source/WebKit/NetworkProcess/storage/SQLiteStorageArea.cpp

```diff
@@ -162,8 +162,18 @@ bool SQLiteStorageArea::prepareDatabase(ShouldCreateIfNotExists shouldCreateIfNo
     FileSystem::makeAllDirectories(FileSystem::parentPath(m_path));
     if (!m_database->open(m_path)) {
         RELEASE_LOG_ERROR(Storage, "SQLiteStorageArea::prepareDatabase failed to open database at '%s'", m_path.utf8().data());
-        m_database = nullptr;
-        return false;
+        if (m_database->lastError() == SQLITE_CORRUPT || m_database->lastError() == SQLITE_NOTADB) {
+            RELEASE_LOG(Storage, "SQLiteStorageArea::prepareDatabase delete database file '%s'", m_path.utf8().data());
+            WebCore::SQLiteFileSystem::deleteDatabaseFile(m_path);
+            if (!m_database->open(m_path)) {
+                RELEASE_LOG_ERROR(Storage, "SQLiteStorageArea::prepareDatabase failed to reopen database after delete at '%s'", m_path.utf8().data());
+                m_database = nullptr;
+                return false;
+            }
+        } else {
+            m_database = nullptr;
+            return false;
+        }
```

Comment:
With `handleDatabaseCorruptionIfNeeded`, we could change this to:
```
bool openResult  = m_database->open(m_path);
if (!openResult && handleDatabaseCorruptionIfNeeded(m_database->lastError())
    openResult = m_database->open(m_path);

if (!openResult) {
    RELEASE_LOG_ERROR(Storage, "SQLiteStorageArea::prepareDatabase failed to open database at '%s'", m_path.utf8().data());
    m_database = nullptr;
    return false;
}
```

Reply:
great optimization, thanks